### PR TITLE
Make master index optional when master is used in other contexts

### PR DESCRIPTION
**Pull request by smarterclayton**
OpenShift provides a default URL at the root that shows the UI. The
provided flag makes loading the index handler optional for now.

### CODE REVIEW

**Comment by bgrant0607** [comment]
Or, right, sorry.
/cc @nikhiljindal

File: pkg/master/master.go [hunk 1]
```diff
@@ -65,17 +65,21 @@ import (
...
+	KubeletClient     client.KubeletClient
+	PortalNet         *net.IPNet
+	EnableLogsSupport bool
+	EnableUISupport   bool
+	// allow downstream consumers to disable swagger
+	EnableSwaggerSupport bool
+	// allow v1beta3 to be conditionally enabled
+	EnableV1Beta3 bool
+	// allow downstream consumers to disable the index route
+	DisableIndex           bool
```
Comment (bgrant0607) [inline pkg/master/master.go]: I'd prefer to make it EnableIndex, for consistency with the other options.

Comment (smarterclayton) [inline pkg/master/master.go]: Ok - I did DisableIndex so I didn't accidentally disable it in all the places.  Will change.

----- Original Message -----

**Comment by smarterclayton** [comment]
I reversed it, it was only a few places.  Agree it looks better.

File: pkg/master/master.go [hunk 2]
```diff
@@ -65,17 +65,21 @@ import (
...
+	EventTTL          time.Duration
+	MinionRegexp      string
+	KubeletClient     client.KubeletClient
+	PortalNet         *net.IPNet
+	EnableLogsSupport bool
+	EnableUISupport   bool
+	// allow downstream consumers to disable swagger
+	EnableSwaggerSupport bool
+	// allow v1beta3 to be conditionally enabled
+	EnableV1Beta3 bool
```
Comment (bgrant0607) [inline pkg/master/master.go]: Did gofmt do this?

**Comment by bgrant0607** [comment]
LGTM, but test failures look real (consistent at least)

**Comment by bgrant0607** [comment]
```
	auth_test.go:313: case {GET /  map[200:true]}
	auth_test.go:326: Expected status one of map[200:true], but got 404
```

File: pkg/master/master.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/master/master.go]: Yeah, comments reset the margin on the right items.

----- Original Message -----

**Comment by smarterclayton** [comment]
I'm going to pretend like I was just validating negative testing here...

----- Original Message -----

**Comment by bgrant0607** [comment]
Rerunning Travis.

**Comment by a-robinson** [comment]
Travis is green. Merging.

**Pull request merged**